### PR TITLE
Show modules open in the results page, if they are open for reading

### PR DESCRIPTION
**Pull request by alalav1**
# Description
If module is open for reading, it is now opened by default in the
exercise results page. Add a badge to inform that this module is for
reading only. Add correct prepositions with dates (Opens on saturday).
Template tag `module_accessible` fixed to work properly, even when
opening time is not set.
Related issue #473 

# Testing

- [ ] I created new unit tests
- [ ] All unit tests pass
- [x] I have tested manually

# Have you updated the README or other relevant documentation?

- [ ] Yes
- [ ] Not yet. I will do it next.
- [x] Not relevant

### CODE REVIEW

**Comment by alalav1** [comment]
I found out that links to the exercises are not functional, if the module closing time is passed, but the late submissions are still allowed. Sadly, the currently used `exercises_open` function cannot be modified, since it is used to separate late and open exercises in the other templates. The `have exercises_been_opened` function could be used.

File: exercise/templatetags/exercise.py [hunk 1]
```diff
@@ -223,8 +223,11 @@ def min_group_size(context):
 
 @register.simple_tag(takes_context=True)
 def module_accessible(context, entry):
-    t = entry.get('reading_opening_time', entry.get('opening_time'))
-    return _is_accessible(context, entry, t)
+    t = entry.get('reading_opening_time')
```
Comment (Mankro) [inline exercise/templatetags/exercise.py]: Does this really change anything? Looks like it does the same stuff as the old code.

**Comment by Mankro** [comment]
Have the exercise links been fixed when late submissions are allowed?

File: exercise/templatetags/exercise.py (hunk 1 above)
Comment (alalav1) [reply exercise/templatetags/exercise.py]: The old code didn't work, if `reading_opening_time` was not set. I think it was maybe the nested get method that caused malfunction. (The outer get method would always find the object even if it was empty.)

Comment (alalav1) [reply exercise/templatetags/exercise.py]: But the logic should be same, and the changed version works as it should.

**Comment by alalav1** [comment]
> Have the exercise links been fixed when late submissions are allowed?

That fix is made in another branch with other late submission related changes.

**Comment by Mankro** [comment]
Are you referring to https://github.com/apluslms/a-plus/pull/475?

**Comment by alalav1** [comment]
Yes.

File: exercise/templates/exercise/_user_results.html [hunk 2]
```diff
@@ -24,14 +24,21 @@
...
+    aria-expanded="{% if accessible %}true{% else %}false{% endif %}" aria-controls="#module{{ module.id }}">
     <h3 class="panel-title">
       {% points_badge module "pull-right" %}
-      {% if not open and not after_open %}
+      {% if not accessible %}
       <span class="badge pull-right">
-        {% trans "Opens" %} {{ module.opening_time }}
+        {% trans "Opens on" %} {{ module.opening_time }}
+      </span>
+      {% elif not open and not after_open %}
```
Comment (Mankro) [inline exercise/templates/exercise/_user_results.html]: `not after_open` would be shorter and semantically identical, right?

File: exercise/templates/exercise/_user_results.html [hunk 3]
```diff
@@ -44,7 +51,7 @@ <h3 class="panel-title">
       {{ module.name|parse_localization }}
     </h3>
   </a>
-  <div class="collapse{% if open and accessible %} in{% endif %}" id="module{{ module.id }}">
```
Comment (Mankro) [inline exercise/templates/exercise/_user_results.html]: Previously, when a module got closed (after the close time), the div was also closed/collapsed. After your change, the div is always open forever after the open time. This should be fixed. The div should be closed after the module close time. This probably affects the `a.panel-heading` element in the top too.

File: exercise/templates/exercise/_user_results.html [hunk 4]
```diff
@@ -24,14 +24,21 @@
...
+      {% if not accessible %}
       <span class="badge pull-right">
-        {% trans "Opens" %} {{ module.opening_time }}
+        {% trans "Opens on" %} {{ module.opening_time }}
+      </span>
+      {% elif not after_open %}
+      <span class="badge pull-right">
+        {% trans "Exercises open on" %} {{ module.opening_time }}
+      </span>
+      <span class="badge badge-success pull-right">
```
Comment (Mankro) [inline exercise/templates/exercise/_user_results.html]: Change the class on the "open for reading" badge from `badge-success` to `badge-info`, please.

**Comment by Mankro** [comment]
I squashed, slightly modified the commit message and force-pushed before merging.

**Pull request merged**